### ISSUE TITLE
A little confusing on how to generate the feature vectors for each app.

### ISSUE THREAD

**Issue by jyxu2015** [jallen89/pikadroid#1]
Hi Allen,

I am now reading your paper and feel very inspired from your idea about combining the entry point to the APIs to further identify the relationship between a sensitive activity and the label of the App. However, after reading the paper and the sourcecodes, I am now a little unsure about my understanding on the generation of feature vector for each app in the training set. 

In your paper section 3.3, it is said that "The feature vector for each a ∈ T is va = ⟨re,s : (e,s) ∈ a⟩, the union of risk values for each (e,s) ∈ a." And in your code, I found the line "features = pd.concat([ben, mal]).fillna(0.0)".

Does it mean that:
(1) given a training set T, for each score r_e,s extracted from T, is there only one value?

(2) for each app in T, it is composed of the union of r_e,s. However, since there are different numbers of r_e,s pairs included in different app. Do you pad "0" to each union to make all the feature vectors in T have an equal length?
- For example:  
       App A will have a feature vector of <6.89, 6.20, 0, ....> if there are only two pairs among all the pairs extract from T existing in A. 
        If there are in total K different pairs extracted from T, the length of the feature vector of Apps in T will all be K, and in the feature vector of A, all the other values in its feature vector are 0 except the two value corresponding to the 2 pairs?

Thank you so much for your help, I am looking forward to your kind reply.

Best Regards,
Jiayun Xu.

**Comment by jallen89** [comment]
Hi Jiayun,

Thank you for the kind comments.  

The feature space for pikaDroid's classification is the set of all entrypoint-api pairs found in the training set. Each dimension in the feature space will represent a potential entrypoint-api pair. When creating the feature vector for each app, if the application does not use a particular entrypoint-api pair, we set the component in the feature vector that represents that entrypoint-api pair to zero. If the app does use the entrypoint-api pair we set the component to the value to r_e,s. 

One final note, We are still finalizing the public release of pikaDroid's source code, which we intend to have finalized by early next year. In some cases, some code may not have been added to the repo yet or lacking proper documentation.

thanks,
Joey
